src/libostree/ostree-impl-system-generator.c

```diff
@@ -264,13 +296,15 @@ _ostree_impl_system_generator (const char *normal_dir, const char *early_dir, co
 
   g_autoptr (GError) otcore_get_ostree_target_error = NULL;
   g_autofree char *ostree_target = NULL;
-  bool is_aboot = false;
-  /* This could happen in CoreOS live environments, where we hackily mock
+  gboolean is_aboot = false;
+  /* If no `ostree=` karg exists, gracefully no-op.
+   * This could happen in CoreOS live environments, where we hackily mock
    * the `ostree=` karg for `ostree-prepare-root.service` specifically, but
    * otherwise that karg doesn't exist on the real command-line. */
   if (!otcore_get_ostree_target (cmdline, &is_aboot, &ostree_target,
                                  &otcore_get_ostree_target_error))
-    return TRUE;
+    errx (EXIT_FAILURE, "Failed to determine ostree target: %s",
```

Comment:
For here, see https://github.com/ostreedev/ostree/pull/3192/files/4a71845b127f8198c66d2fd2f024db8b01afb618#r1499795034.